### PR TITLE
fix: Add einops as a core dependency

### LINKED ISSUE #39811: Missing einops dependency causing ModuleNotFoundError
### System Info

Copy-and-paste the text below in your GitHub issue and FILL OUT the two last points.
- `transformers` version: 4.53.2
- Platform: Linux-4.19.90-vhulk2211.3.0.h1543.eulerosv2r10.aarch64-aarch64-with-glibc2.35
- Python version: 3.11.13
- Huggingface_hub version: 0.34.3
- Safetensors version: 0.5.3
- Accelerate version: not installed
- Accelerate config: not found
- DeepSpeed version: not installed
- PyTorch version (accelerator?): 2.6.0+cpu (NPU)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: yes
- Using NPU in script?: yes
- NPU type: Ascend910B4
- CANN version: 8.2.RC1.alpha003

### Who can help?

@ivarflakstad 

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

Background: running SGLang on Ascend NPU

1. Run with ascend/cann:8.2.rc1.alpha003-910b-ubuntu22.04-py3.11 docker image
2. Install deps: `pip install torch==2.6.0 torchvision==0.21.0 --index-url https://download.pytorch.org/whl/cpu && pip install torch_npu==2.6.0`
3. Clone sglang and run `pip install -e "python[srt_npu]"`
4. Check `transformers env`

Image

### Expected behavior

import transformers successfully

### PR DESCRIPTION
**Pull request by Hashbrownsss**
# What does this PR do?

This PR addresses the `ModuleNotFoundError` for the `einops` library by adding it as a core dependency.

It fixes issue #39811, where certain parts of the `transformers` library, specifically in the flash attention code, were trying to import `einops` without it being a declared dependency. This bug prevented users from successfully running models that rely on this library.

By adding `einops` to `setup.py`, this PR ensures that the library is automatically installed for all users, making the `transformers` package more robust and reliable.

Fixes #39811

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case. This PR fixes issue #39811.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
    > This change is a dependency fix and does not require a documentation update.
- [ ] Did you write any new necessary tests?
    > This change is a dependency fix and does not require a new test. The existing test suite was used to confirm that the `einops` dependency is now correctly installed.

## Who can review?

@ArthurZucker 
@ivarflakstad 
@iforgetmyname

### CODE REVIEW

**Review by ArthurZucker — COMMENTED** [review]
Thanks! Tho I think this should be added to the `npu` extra instead!

**Comment by Hashbrownsss** [comment]
Hey! Thanks for the quick feedback. Although i couldnt find a dedicated npu extra. ive updated the PR to add einops to dev-torch instead, as it seems like the most apt place for the dependency I hope that works!

**Comment by vasqu** [comment]
It seems like the einops dependency is only due to the un-/padding functions. It would be best to just rewrite them with normal torch and/or use the fa3 equivalents in the fa modeling utils file - no need for the einops dependency.

**Comment by vasqu** [comment]
#40002 will probably remove the einops dependency but will keep you updated if not

**Comment by Hashbrownsss** [comment]
That sounds good! I'll close this PR and keep an  eye on #40002
